### PR TITLE
Action model

### PR DESCRIPTION
**Pull request by benfrancis**
This creates a model layer for Actions and moves business logic out of the controller and into the model where it should be.

Note: This is top level actions on the gateway, not actions on Things which are part of the Web Thing API and will come later.

### CODE REVIEW

File: models/action.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+'use strict';
+
+var Actions = require('../models/actions');
+
+var Action = function(name) {
+  this.id = Actions.generateId();
+  this.name = name;
+  this.href = '/actions/' + this.id;
```
Comment (benfrancis) [inline models/action.js]: Yep, good idea! This shouldn't be hard coded in the model.

File: models/actions.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  add: function(action) {
+    var id = action.id;
+    this.actions[id] = action;
+    switch(action.name) {
+      case 'pair':
+        AdapterManager.addNewThing().then(function(thing) {
+          Things.handleNewThing(thing);
+        }).catch(function(error) {
+          console.error('Error trying to add new thing ' + error);
```
Comment (benfrancis) [inline models/actions.js]: Good point.

File: config.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+'use strict';
+
+exports.THINGS_PATH = '/things';
+exports.NEW_THINGS_PATH = '/new_things';
+exports.ADAPTERS_PATH = '/adapters';
+exports.ACTIONS_PATH = '/actions';
+exports.STATIC_PATH = 'static';
+exports.DEBUG_PATH = '/debug';
```
Comment (dhylands) [inline config.js]: There is also a config directory. Currently, it just has the gpio config. Perhaps this should be a file in the config directory, and have index.js

**Comment by benfrancis** [comment]
Done, thanks.

**Pull request merged**